adds login view and route, updates user db to hashed passwords

hy-js opened the pull request:
@GregBaughDev  #Action
Hi Greg, - Ive added the login route and view. Also changed user DB to hash passwords to confirm bcrypt compare is working. added unhashed-passwords file for reference and cleaned up the README a little.

I'm currently plugging in correct values to the login form for quick testing and have tried out bootstrap but both of these will obviously be replaced later.

Having issues with my .env file but reverted the conn.js back to normal.

GregBaughDev commented:
@hy-js #Action
Hi Rhys! This looks great!
I had a couple of thoughts and wanted to shoot over some feedback for consideration.

- login.js - Lines 33 to 47
From what I've read with user logins we want to avoid letting people know if an account exists for the user. i.e. "incorrect password for this account" as it lets hackers know that an email address is valid but the password is wrong. 
I had a thought for a small refactor below:
```
db.oneOrNone('SELECT * FROM users WHERE email = $1 [req.body.email.toLowerCase()]
.then(user => 
if(user){
const validPassword = await bcrypt.compare(req.body.password, user.password);
if(validPassword)
{ // log user in}}
else {
// Send details not found
} 
```

- Session Middleware
Thanks so much for setting up the middleware 🙌 - I think we are going over this tomorrow but if we include this in our index.js it is accessible to all our routes. I've cloned the code and done some experimenting - so in the login.js file, if the user details are correct I set ```req.session.loggedIn = true```(this was set in login.js after the password and email are checked). On the home route I added a console.log to see if the information persists: 
``` // Home route
app.get('/', async (req, res) => {
    const allScheds = await db.any("SELECT * FROM schedules LEFT JOIN users ON schedules.user_id = users.user_id")
    console.log(req.session.loggedIn)
    res.render('index', {allScheds})
})``` and it did ('true' appeared on the console when I went to our root file)
So then I tried the following:
``` app.get('/sessiontest', (req, res) => {
    req.session.loggedIn ? res.send('logged in') : res.send('not logged in')
})```
And the conditional test worked. So with this we can now set 'logged in' to true so users can access all parts of the site. The only issue is that the length of the session is set to a minute (the maxAge value in the session object) which meant that after a minute we were logged back out. 

We can include the session conditionals at a later stage if you like? Thanks so much for doing all this! Let me know if anything's unclear and let me know if you want to have a chat tomorrow about anything?

Let me know your thoughts and I'm happy to merge the code as is or I can wait if you wanted to make any changes?

hy-js commented:
@GregBaughDev #Action
Hi Greg, Thanks so much for your feedback. I have made the following changes:

- login.js 
Removed user confirmation and replaced with temporary ambiguous "Invalid Details" console.log.
Updated logic to [guard clause](https://maximegel.medium.com/what-are-guard-clauses-and-how-to-use-them-350c8f1b6fd2) structure for hopefully cleaner code.
Added view logic to only display login form if not logged in,|

- Session middleware
Added redirect to log in view if not logged in. 
Extended cookie time to 1 hour.

There is still a bit of work to do on the session cookies, hopefully with some revision after class this week I can add log out and cleaner redirect functionality.

GregBaughDev commented:
Hey Rhys! Amazing and so clean, thank you for doing all this!
